\name{pwLines}
\Rdversion{1.1}
\alias{pwLines}
\docType{data}
\title{
A representation of Bild's HMEC lines for pathway signature identification
}
\description{
A representation of Bild's HMEC lines for pathway signature identification
}
\usage{data(pwLines)}
\format{
  The format is:
Formal class 'ExpressionSet' [package "Biobase"] with 6 slots
  ..@ assayData        :<environment: 0x192e1dc> 
  ..@ phenoData        :Formal class 'AnnotatedDataFrame' [package "Biobase"] with 4 slots
  .. .. ..@ varMetadata      :'data.frame':	28 obs. of  1 variable:
  .. .. .. ..$ labelDescription: chr [1:28] NA NA NA NA ...
  .. .. ..@ data             :'data.frame':	55 obs. of  28 variables:
  .. .. .. ..$ gsmid                  : chr [1:55] "GSM70597" "GSM70598" "GSM70599" "GSM70600" ...
  .. .. .. ..$ gsmlabel               : Factor w/ 55 levels "BCAT-8","BCAT-9",..: 19 21 22 23 24 25 26 27 28 20 ...
  .. .. .. ..$ perturb                : chr [1:55] "GFP" "GFP" "GFP" "GFP" ...
  .. .. .. ..$ title                  : Factor w/ 55 levels "BCAT-8","BCAT-9",..: 19 21 22 23 24 25 26 27 28 20 ...
  .. .. .. .. ..- attr(*, "names")= chr [1:55] "V2" "V3" "V4" "V5" ...
  .. .. .. ..$ geo_accession          : Factor w/ 55 levels "GSM70597","GSM70598",..: 1 2 3 4 5 6 7 8 9 10 ...
  .. .. .. .. ..- attr(*, "names")= chr [1:55] "V2" "V3" "V4" "V5" ...
  .. .. .. ..$ status                 : Factor w/ 1 level "Public on Nov 07 2005": 1 1 1 1 1 1 1 1 1 1 ...
  .. .. .. .. ..- attr(*, "names")= chr [1:55] "V2" "V3" "V4" "V5" ...
  .. .. .. ..$ submission_date        : Factor w/ 1 level "Aug 17 2005": 1 1 1 1 1 1 1 1 1 1 ...
  .. .. .. .. ..- attr(*, "names")= chr [1:55] "V2" "V3" "V4" "V5" ...
  .. .. .. ..$ last_update_date       : Factor w/ 1 level "Nov 07 2005": 1 1 1 1 1 1 1 1 1 1 ...
  .. .. .. .. ..- attr(*, "names")= chr [1:55] "V2" "V3" "V4" "V5" ...
  .. .. .. ..$ type                   : Factor w/ 1 level "RNA": 1 1 1 1 1 1 1 1 1 1 ...
  .. .. .. .. ..- attr(*, "names")= chr [1:55] "V2" "V3" "V4" "V5" ...
  .. .. .. ..$ channel_count          : Factor w/ 1 level "1": 1 1 1 1 1 1 1 1 1 1 ...
  .. .. .. .. ..- attr(*, "names")= chr [1:55] "V2" "V3" "V4" "V5" ...
  .. .. .. ..$ source_name_ch1        : Factor w/ 1 level "mammary epithelial cells expressing oncogenes (or GFP control)": 1 1 1 1 1 1 1 1 1 1 ...
  .. .. .. .. ..- attr(*, "names")= chr [1:55] "V2" "V3" "V4" "V5" ...
  .. .. .. ..$ organism_ch1           : Factor w/ 1 level "Homo sapiens": 1 1 1 1 1 1 1 1 1 1 ...
  .. .. .. .. ..- attr(*, "names")= chr [1:55] "V2" "V3" "V4" "V5" ...
  .. .. .. ..$ characteristics_ch1    : Factor w/ 1 level "mammary epithelial cells expressing oncogenes (or GFP control)": 1 1 1 1 1 1 1 1 1 1 ...
  .. .. .. .. ..- attr(*, "names")= chr [1:55] "V2" "V3" "V4" "V5" ...
  .. .. .. ..$ molecule_ch1           : Factor w/ 1 level "total RNA": 1 1 1 1 1 1 1 1 1 1 ...
  .. .. .. .. ..- attr(*, "names")= chr [1:55] "V2" "V3" "V4" "V5" ...
  .. .. .. ..$ label_ch1              : Factor w/ 1 level "biotin": 1 1 1 1 1 1 1 1 1 1 ...
  .. .. .. .. ..- attr(*, "names")= chr [1:55] "V2" "V3" "V4" "V5" ...
  .. .. .. ..$ description            : Factor w/ 1 level "RNA was extracted from human mammary epithelial cells expressing oncogenes (or GFP control)": 1 1 1 1 1 1 1 1 1 1 ...
  .. .. .. .. ..- attr(*, "names")= chr [1:55] "V2" "V3" "V4" "V5" ...
  .. .. .. ..$ data_processing        : Factor w/ 1 level "Affymetrix Microarray Suite version 5.0": 1 1 1 1 1 1 1 1 1 1 ...
  .. .. .. .. ..- attr(*, "names")= chr [1:55] "V2" "V3" "V4" "V5" ...
  .. .. .. ..$ platform_id            : Factor w/ 1 level "GPL570": 1 1 1 1 1 1 1 1 1 1 ...
  .. .. .. .. ..- attr(*, "names")= chr [1:55] "V2" "V3" "V4" "V5" ...
  .. .. .. ..$ contact_name           : Factor w/ 1 level "Joseph,R.,Nevins": 1 1 1 1 1 1 1 1 1 1 ...
  .. .. .. .. ..- attr(*, "names")= chr [1:55] "V2" "V3" "V4" "V5" ...
  .. .. .. ..$ contact_department     : Factor w/ 1 level "IGSP": 1 1 1 1 1 1 1 1 1 1 ...
  .. .. .. .. ..- attr(*, "names")= chr [1:55] "V2" "V3" "V4" "V5" ...
  .. .. .. ..$ contact_institute      : Factor w/ 1 level "Duke University": 1 1 1 1 1 1 1 1 1 1 ...
  .. .. .. .. ..- attr(*, "names")= chr [1:55] "V2" "V3" "V4" "V5" ...
  .. .. .. ..$ contact_address        : Factor w/ 1 level "2133 CIEMAS, 101 Science Dr.": 1 1 1 1 1 1 1 1 1 1 ...
  .. .. .. .. ..- attr(*, "names")= chr [1:55] "V2" "V3" "V4" "V5" ...
  .. .. .. ..$ contact_city           : Factor w/ 1 level "Durham": 1 1 1 1 1 1 1 1 1 1 ...
  .. .. .. .. ..- attr(*, "names")= chr [1:55] "V2" "V3" "V4" "V5" ...
  .. .. .. ..$ contact_state          : Factor w/ 1 level "NC": 1 1 1 1 1 1 1 1 1 1 ...
  .. .. .. .. ..- attr(*, "names")= chr [1:55] "V2" "V3" "V4" "V5" ...
  .. .. .. ..$ contact_zip/postal_code: Factor w/ 1 level "27708": 1 1 1 1 1 1 1 1 1 1 ...
  .. .. .. .. ..- attr(*, "names")= chr [1:55] "V2" "V3" "V4" "V5" ...
  .. .. .. ..$ contact_country        : Factor w/ 1 level "USA": 1 1 1 1 1 1 1 1 1 1 ...
  .. .. .. .. ..- attr(*, "names")= chr [1:55] "V2" "V3" "V4" "V5" ...
  .. .. .. ..$ supplementary_file     : Factor w/ 1 level "NONE": 1 1 1 1 1 1 1 1 1 1 ...
  .. .. .. .. ..- attr(*, "names")= chr [1:55] "V2" "V3" "V4" "V5" ...
  .. .. .. ..$ data_row_count         : Factor w/ 1 level "54675": 1 1 1 1 1 1 1 1 1 1 ...
  .. .. .. .. ..- attr(*, "names")= chr [1:55] "V2" "V3" "V4" "V5" ...
  .. .. ..@ dimLabels        : chr [1:2] "sampleNames" "sampleColumns"
  .. .. ..@ .__classVersion__:Formal class 'Versions' [package "Biobase"] with 1 slots
  .. .. .. .. ..@ .Data:List of 1
  .. .. .. .. .. ..$ : int [1:3] 1 1 0
  ..@ featureData      :Formal class 'AnnotatedDataFrame' [package "Biobase"] with 4 slots
  .. .. ..@ varMetadata      :'data.frame':	16 obs. of  3 variables:
  .. .. .. ..$ Column          : chr [1:16] "ID" "GB_ACC" "SPOT_ID" "Species Scientific Name" ...
  .. .. .. ..$ Description     : Factor w/ 15 levels "","A gene symbol, when one is available (from UniGene).",..: 3 5 15 13 12 1 11 1 10 14 ...
  .. .. .. ..$ labelDescription: chr [1:16] NA NA NA NA ...
  .. .. ..@ data             :'data.frame':	54675 obs. of  16 variables:
  .. .. .. ..$ ID                              : Factor w/ 54676 levels "1007_s_at","1053_at",..: 1 2 3 4 5 6 7 8 9 10 ...
  .. .. .. ..$ GB_ACC                          : Factor w/ 51277 levels "AF276507","AK074161",..: 82 21 83 85 17 16 84 87 19 15 ...
  .. .. .. ..$ SPOT_ID                         : chr [1:54675] NA NA NA NA ...
  .. .. .. ..$ Species.Scientific.Name         : Factor w/ 2 levels "Homo sapiens",..: 1 1 1 1 1 1 1 1 1 1 ...
  .. .. .. ..$ Annotation.Date                 : Factor w/ 2 levels "Mar 11, 2009",..: 1 1 1 1 1 1 1 1 1 1 ...
  .. .. .. ..$ Sequence.Type                   : Factor w/ 4 levels "Consensus sequence",..: 2 2 2 2 2 2 2 2 2 2 ...
  .. .. .. ..$ Sequence.Source                 : Factor w/ 3 levels "Affymetrix Proprietary Database",..: 1 2 1 2 1 2 1 1 2 1 ...
  .. .. .. ..$ Target.Description              : Factor w/ 51843 levels "J02843 /FEATURE=cds /DEFINITION=HUMCYPIIE Human cytochrome P450IIE1 (ethanol-inducible) gene, complete cds",..: 8 7 9 11 3 2 10 13 5 1 ...
  .. .. .. ..$ Representative.Public.ID        : Factor w/ 51339 levels "AF276507","AK074161",..: 82 21 83 85 17 16 84 87 19 15 ...
  .. .. .. ..$ Gene.Title                      : Factor w/ 23061 levels "ADAM metallopeptidase domain 32",..: 34 59 42 54 41 76 72 58 19 33 ...
  .. .. .. ..$ Gene.Symbol                     : Factor w/ 20828 levels "ADAM32","AFG3L1",..: 26 59 40 50 38 75 68 54 18 25 ...
  .. .. .. ..$ ENTREZ_GENE_ID                  : Factor w/ 20828 levels "10406","11078",..: 63 59 51 64 48 62 61 3 60 24 ...
  .. .. .. ..$ RefSeq.Transcript.ID            : Factor w/ 20082 levels "","NM_000409",..: 17 22 19 26 2 25 24 32 23 4 ...
  .. .. .. ..$ Gene.Ontology.Biological.Process: Factor w/ 8891 levels "","0000122 // negative regulation of transcription from RNA polymerase II promoter // inferred from electronic annotation /// 0006"| __truncated__,..: 20 11 36 8 39 18 5 23 35 49 ...
  .. .. .. ..$ Gene.Ontology.Cellular.Component: Factor w/ 6677 levels "","0000151 // ubiquitin ligase complex // inferred from electronic annotation /// 0005634 // nucleus // inferred from electronic a"| __truncated__,..: 43 21 1 20 44 1 18 28 8 36 ...
  .. .. .. ..$ Gene.Ontology.Molecular.Function: Factor w/ 8929 levels "","0000166 // nucleotide binding // inferred from electronic annotation /// 0001784 // phosphotyrosine binding // inferred from el"| __truncated__,..: 9 5 10 23 47 7 22 38 41 37 ...
  .. .. ..@ dimLabels        : chr [1:2] "featureNames" "featureColumns"
  .. .. ..@ .__classVersion__:Formal class 'Versions' [package "Biobase"] with 1 slots
  .. .. .. .. ..@ .Data:List of 1
  .. .. .. .. .. ..$ : int [1:3] 1 1 0
  ..@ experimentData   :Formal class 'MIAME' [package "Biobase"] with 13 slots
  .. .. ..@ name             : chr ""
  .. .. ..@ lab              : chr ""
  .. .. ..@ contact          : chr ""
  .. .. ..@ title            : chr ""
  .. .. ..@ abstract         : chr ""
  .. .. ..@ url              : chr ""
  .. .. ..@ pubMedIds        : chr ""
  .. .. ..@ samples          : list()
  .. .. ..@ hybridizations   : list()
  .. .. ..@ normControls     : list()
  .. .. ..@ preprocessing    : list()
  .. .. ..@ other            : list()
  .. .. ..@ .__classVersion__:Formal class 'Versions' [package "Biobase"] with 1 slots
  .. .. .. .. ..@ .Data:List of 1
  .. .. .. .. .. ..$ : int [1:3] 1 0 0
  ..@ annotation       : chr "hgu133plus2.db"
  ..@ .__classVersion__:Formal class 'Versions' [package "Biobase"] with 1 slots
  .. .. ..@ .Data:List of 4
  .. .. .. ..$ : int [1:3] 2 10 0
  .. .. .. ..$ : int [1:3] 2 3 11
  .. .. .. ..$ : int [1:3] 1 1 0
  .. .. .. ..$ : int [1:3] 1 0 0
}
\details{
%%  ~~ If necessary, more details than the __description__ above ~~
}
\source{
%%  ~~ reference to a publication or URL from which the data were obtained ~~
}
\references{
%%  ~~ possibly secondary sources and usages ~~
}
\examples{
data(pwLines)
## maybe str(pwLines) ; plot(pwLines) ...
}
\keyword{datasets}
